integrated_channels/moodle/migrations/0028_auto_20230928_1530.py

```diff
@@ -0,0 +1,59 @@
+# Generated by Django 3.2.20 on 2023-09-28 15:30
+
+from django.db import migrations
+from integrated_channels.utils import dummy_reverse
+import fernet_fields.fields
+
+
+def populate_decrypted_fields(apps, schema_editor):
+    """
+    Populates the encryption fields with the data previously stored in database.
+    """
+    MoodleEnterpriseCustomerConfiguration = apps.get_model('moodle', 'MoodleEnterpriseCustomerConfiguration')
+
+    for moodle_enterprise_configuration in MoodleEnterpriseCustomerConfiguration.objects.all():
+        moodle_enterprise_configuration.decrypted_username = moodle_enterprise_configuration.username
+        moodle_enterprise_configuration.decrypted_password = moodle_enterprise_configuration.password
+        moodle_enterprise_configuration.decrypted_token = moodle_enterprise_configuration.token
+        moodle_enterprise_configuration.save()
```

Comment:
[suggestion] I think this part should be covered in a separate migration. As per the docs:

> Add the new encrypted field with a different name and initialize its values as null, otherwise decryption will be attempted before anything has been encrypted.

We should initially add new field and save nulls, then the second migration can take care of copying values so decryption only takes place after encryption.

Reply:
@sameenfatima78 it can be a better strategy